core/src/main/kotlin/cash/grammar/kotlindsl/utils/Whitespace.kt

```diff
@@ -112,6 +114,45 @@ public object Whitespace {
     return tokens.getHiddenTokensToRight(before.tokenIndex, KotlinLexer.WHITESPACE)
   }
 
+  /**
+   * Returns the indentation in this input, based on the assumption that the first indentation
+   * discovered is the common indent level. If no indent discovered (which could happen if this
+   * input contains only top-level statements), defaults to [min].
+   */
+  public fun computeIndent(
```

Comment:
I don't really like this function because it feels like a poor man's regex, and the whole point of creating a parse tree is avoiding regex -- nonetheless, dealing with whitespace has proven to be a huge pain, this works, and I think we ought to be able to further improve it.